src/main/java/io/divolte/server/filesinks/FileFlusher.java

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright 2017 GoDataDriven B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.divolte.server.filesinks;
+
+import static io.divolte.server.processing.ItemProcessor.ProcessingDirective.*;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.MoreObjects;
+
+import io.divolte.server.AvroRecordBuffer;
+import io.divolte.server.config.FileStrategyConfiguration;
+import io.divolte.server.filesinks.FileManager.DivolteFile;
+import io.divolte.server.processing.Item;
+import io.divolte.server.processing.ItemProcessor;
+
+@ParametersAreNonnullByDefault
+public class FileFlusher implements ItemProcessor<AvroRecordBuffer> {
```

Comment:
Perhaps it could, but I don't see the immediate gain compared to better naming of some vars. Altogether, the `FileFlusher` makes heavy use of internal mutable state, including recorded timings and the observed healthiness of the underlying filesystem. I chose to encode these using primitives where they didn't confuse too much. In a previous iteration, the current file handle was also nullable (instead of using `Optional` as it is now) and there was a separate boolean keeping track of observer filesystem health, but that had understandability issues.